### PR TITLE
Move BilinearTensorProduct OP to phi

### PR DESCRIPTION
**Pull request by From00**
### PR types
Function optimization

### PR changes
OPs

### Describe

Move BilinearTensorProduct OP to phi.

### CODE REVIEW

File: paddle/phi/infermeta/multiary.cc [hunk 1]
```diff
@@ -18,6 +18,71 @@ limitations under the License. */
...
+                          "The Input(Bias) must be a 2-D tensor with "
+                          "the 2nd dimension fixed to 1 (a row vector)."));
+    PADDLE_ENFORCE_EQ(bias_dims[1],
+                      weight_dims[0],
+                      errors::InvalidArgument(
+                          "The second dimension of input(Bias) must be equal "
+                          "to the first dimension of the input(Weight)."));
+  }
+
+  out->set_dims({x_dims[0], weight_dims[0]});
```
Comment (MingMingShangTian) [inline paddle/phi/infermeta/multiary.cc:82]: 这里out也需要设置一下dtype

File: paddle/phi/kernels/bilinear_tensor_product_grad_kernel.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/phi/core/dense_tensor.h"
+#include "paddle/utils/optional.h"
```
Comment (MingMingShangTian) [inline paddle/phi/kernels/bilinear_tensor_product_grad_kernel.h]: 这里没用到optional.h 头文件

File: paddle/phi/kernels/impl/bilinear_tensor_product_grad_kernel_impl.h [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/phi/core/dense_tensor.h"
+#include "paddle/phi/kernels/funcs/blas/blas.h"
+#include "paddle/phi/kernels/funcs/eigen/common.h"
+#include "paddle/utils/optional.h"
```
Comment (MingMingShangTian) [inline paddle/phi/kernels/impl/bilinear_tensor_product_grad_kernel_impl.h]: 也没用到 optional.h

File: paddle/phi/infermeta/multiary.cc (hunk 1 above)
Comment (From00) [reply paddle/phi/infermeta/multiary.cc:82]: done

File: paddle/phi/kernels/impl/bilinear_tensor_product_grad_kernel_impl.h (hunk 3 above)
Comment (From00) [reply paddle/phi/kernels/impl/bilinear_tensor_product_grad_kernel_impl.h]: done

File: paddle/phi/kernels/bilinear_tensor_product_grad_kernel.h (hunk 2 above)
Comment (From00) [reply paddle/phi/kernels/bilinear_tensor_product_grad_kernel.h]: done

**Review by MingMingShangTian — APPROVED** [review]
LGTM

**Pull request merged**